exporter/collector/metrics.go

```diff
@@ -1200,7 +1235,12 @@ func mapMetricPointKind(m pmetric.Metric) (metricpb.MetricDescriptor_MetricKind,
 			kind = metricpb.MetricDescriptor_CUMULATIVE
 		}
 		if m.Sum().DataPoints().Len() > 0 {
-			typ = metricPointValueType(m.Sum().DataPoints().At(0).ValueType())
+			_, supportedType := me.convertMetricKindToBoolIfSupported(m.Sum().DataPoints().At(0), kind, m.Unit())
```

Comment:
Right, that's a good catch. We can/should enforce only OTLP Gauges.

Reply:
The `kind` parameter is passed to provide a sanity check warning - in case the function is called on a metric kind other than `gauge`. So I think its better to keep it. 

If we move the `kind` parameter out of the function then we need to do that check before calling the function and then logging appropriate message.